includes/admin/class-wc-admin-profile.php

```diff
@@ -154,7 +154,7 @@ public function get_customer_meta_fields() {
 		 * @param WP_User $user
 		 */
 		public function add_customer_meta_fields( $user ) {
-			if ( ! current_user_can( 'manage_woocommerce' ) ) {
+			if ( ! current_user_can( 'manage_woocommerce' ) && ! apply_filters( 'woocommerce_current_user_can_edit_customer_meta_fields', false, $user ) ) {
```

Comment:
Do you mean something like ...

```
$user_meta_permission = apply_filters( 'woocommerce_customer_meta_fields_permission', 'manage_woocommerce' );

if ( current_user_can( $user_meta_permission ) ) {
    // ...
}
```

My particular use case would allow for this, though I'm not sure if it would be too restrictive to users who want to tie this option to something not related to roles.

I can make the adjustment or leave it as is, let me know!
John